contracts/RMRK/extension/tokenProperties/IRMRKPropertiesAccessControl.sol

```diff
@@ -0,0 +1,157 @@
+     * @return The `bytes` type.
+     * @return The `uint` type.
+     * @return The `bool` type.
+     */
+    enum ParameterType {
+        STRING,
+        ADDRESS,
+        BYTES,
+        UINT,
+        BOOL
+    }
+
+    /**
+     * @notice A list of supported access types.
+     * @return The `Issuer` type, where only the issuer can manage the parameter.
+     * @return The `Collaborator` type, where only the collaborators can manage the parameter.
+     * @return The `IssuerOrCollaborator` type, where only the issuer or collaborators can manage the parameter.
+     * @return The `TokenOwner` type, where only the token owner can manage the parameters of their tokens.
+     * @return The `SpecificAddress` type, where only specific addresses can manage the parameter.
+     */
+    enum AccessType {
+        Issuer,
+        Collaborator,
+        IssuerOrCollaborator,
+        TokenOwner,
+        SpecificAddress
+    }
+
+    /**
+     * @notice Used to noitfy listeners that the access control settings for a specific parameter have been updated.
+     * @param collection Address of the collection.
+     * @param parameterType The type of parameter for which the access control settings have been updated.
+     * @param parameterId The ID of the parameter for which the access control settings have been updated.
+     * @param accessType The AccessType of the parameter for which the access control settings have been updated.
+     * @param collaboratorAddresses The array of collaborator addresses that have been updated.
+     * @param collaboratorAddressAccess The boolean array of the access values for the collaborator addresses.
+     * @param specificAddresses The array of specific addresses that have been updated.
+     * @param specificAddressAccess The boolean array of the access values for the specific addresses.
+     */
+    event AccessControlUpdated(
+        address indexed collection,
+        ParameterType parameterType,
+        uint256 parameterId,
+        AccessType accessType,
+        address[] collaboratorAddresses,
+        bool[] collaboratorAddressAccess,
+        address[] specificAddresses,
+        bool[] specificAddressAccess
+    );
+
+    /**
+     * @notice Used to register a collection to use the RMRK token properties repository.
+     * @dev  If the collection does not implement the Ownable interface, the `useOwnable` value must be set to `false`.
+     * @param collection The address of the collection that will use the RMRK token properties repository.
+     * @param issuer The address of the issuer of the collection.
+     * @param useOwnable The boolean value to indicate if the collection implements the Ownable interface and whether it
+     *  should be used to validate that the caller is the issuer (`true`) or to use the manually set issuer address
+     *  (`false`).
+     */
+    function registerAccessControl(
+        address collection,
+        address issuer,
+        bool useOwnable
```

Comment:
I would simply not support anything not using ownable. The alternative way, if I understand correctly, would allow anyone to claim ownership of some collection's properties

Reply:
I'd leave it. I think it's beneficial to have the ability for all the collections to use the repository. Yes, anyone can utilise the repository to claim the issuer role if the collection doesn't implement Ownable, but I don't see this as a big issue.

1. Only using Ownable bars any collection that doesn't have `owner()` implemented.
2. Allowing to claim collections that don't have `owner()` is inclusive of all collections. There is a risk that a 3rd party could claim it (this would mean that the legitimate issuer can't use it, which is essentially the same as option 1), but there is also the ability to use this if the collection uses some other kind of access control or if it is self-owned.

I think option 2 allows for more inclusion.. 🤔